src/event.processor.ts

```diff
@@ -0,0 +1,142 @@
+import axios from "axios";
+
+export class EventProcessor {
+  private options: EventProcessorOptions = new EventProcessorOptions();
+
+  async start(options: EventProcessorOptions) {
+    this.options = options;
+
+    if (!options.emitterAddresses && !options.eventIdentifiers) {
+      throw new Error(`No emitter address or identifier set. Could not resolve all the events without filters.`);
+    }
+
+    await this.startProcess(options);
+  }
+
+  private async startProcess(options: EventProcessorOptions) {
+    const lastTimestampFunc = options.getLastProcessedTimestamp;
+    if (!lastTimestampFunc) {
+      throw new Error(`Undefined function for getting the last processed timestamp`);
+    }
+
+    const maxHeight = await lastTimestampFunc();
+    if (maxHeight === undefined) {
+      throw new Error(`Cannot get the last processed timestamp via the provided getLastProcessedTimestamp()`);
+    }
+
+    await this.callElasticsearchEvents(maxHeight);
+  }
+
+  async callElasticsearchEvents(lastProcessedTimestamp: number): Promise<void> {
+    const url = `${this.options.elasticUrl}/events/_search?scroll=${this.options.scrollTimeout}`;
+    const elasticQuery = this.generateElasticsearchQuery(lastProcessedTimestamp);
+    const result = await axios.post(url, elasticQuery);
+
+    if (!result.data || !result.data.hits || !result.data.hits || !result.data.hits.hits) {
+      return;
+    }
+
+    const elasticEvents = result.data.hits.hits;
+    const events = elasticEvents.map((e: { _source: any; }) => e._source);
+    await this.handleElasticEvents(events);
+
+    const scrollId = result.data._scroll_id;
+    while (true) {
```

Comment:
maybe you can use`while (scrollId)` insead of `while (true)` because provides a clear exit condition for the loop based on the presence of the `scrollId`